### ISSUE TITLE
Adding "other Packages" and "Testing FlowUnsteady" Fails

### ISSUE THREAD

**Issue by philippmandl** [byuflowlab/FLOWUnsteady#74]

**Comment by EdoAlvarezR** [comment]
@philippmandl, could you please comment on how you were able to resolve the issue?

**Comment by philippmandl** [comment]
> there was a bug in the python-path to anaconda 3 which wasn't found by the system.
> 
> An additional failure comes up when I try to install the software on Scientific Linux 7.9:
> 
>     * the error message is linked to the ] test FLOWExaFMM execution-
> 
> 
> **Error-Message:**
> 
> pkg> test FLOWExaFMM Testing FLOWExaFMM Status `/tmp/jl_aVi9OH/Project.toml` [1f15a43c] CxxWrap v0.11.2 [a07d1f4e] FLOWExaFMM v2.1.0 `/local/disk1/pmandl/01_Software/FLOWExaFMM` [8dfed614] Test `@stdlib/Test` Status `/tmp/jl_aVi9OH/Manifest.toml` [1f15a43c] CxxWrap v0.11.2 [a07d1f4e] FLOWExaFMM v2.1.0 `/local/disk1/pmandl/01_Software/FLOWExaFMM` [692b3bcd] JLLWrappers v1.4.1 [1914dd2f] MacroTools v0.5.10 [21216c6a] Preferences v1.4.0 [3eaa8342] libcxxwrap_julia_jll v0.8.8+1 [0dad84c5] ArgTools `@stdlib/ArgTools` [56f22d72] Artifacts `@stdlib/Artifacts` [2a0f44e3] Base64 `@stdlib/Base64` [ade2ca70] Dates `@stdlib/Dates` [f43a241f] Downloads `@stdlib/Downloads` [b77e0a4c] InteractiveUtils `@stdlib/InteractiveUtils` [b27032c2] LibCURL `@stdlib/LibCURL` [76f85450] LibGit2 `@stdlib/LibGit2` [8f399da3] Libdl `@stdlib/Libdl` [56ddb016] Logging `@stdlib/Logging` [d6f4376e] Markdown `@stdlib/Markdown` [ca575930] NetworkOptions `@stdlib/NetworkOptions` [44cfe95a] Pkg `@stdlib/Pkg` [de0858da] Printf `@stdlib/Printf` [3fa0cd96] REPL `@stdlib/REPL` [9a3f8284] Random `@stdlib/Random` [ea8e919c] SHA `@stdlib/SHA` [9e88b42a] Serialization `@stdlib/Serialization` [6462fe0b] Sockets `@stdlib/Sockets` [fa267f1f] TOML `@stdlib/TOML` [a4e569a6] Tar `@stdlib/Tar` [8dfed614] Test `@stdlib/Test` [cf7118a7] UUIDs `@stdlib/UUIDs` [4ec0a83e] Unicode `@stdlib/Unicode` [deac9b47] LibCURL_jll `@stdlib/LibCURL_jll` [29816b5a] LibSSH2_jll `@stdlib/LibSSH2_jll` [c8ffd9c3] MbedTLS_jll `@stdlib/MbedTLS_jll` [14a3606d] MozillaCACerts_jll `@stdlib/MozillaCACerts_jll` [83775a58] Zlib_jll `@stdlib/Zlib_jll` [8e850ede] nghttp2_jll `@stdlib/nghttp2_jll` [3f19e933] p7zip_jll `@stdlib/p7zip_jll` Precompiling project... ✗ FLOWExaFMM 0 dependencies successfully precompiled in 1 seconds (8 already precompiled) 1 dependency errored. To see a full report either run `import Pkg; Pkg.precompile()` or load the package Testing Running tests... /local/disk1/pmandl/01_Software/julia-1.6.7/bin/julia: symbol lookup error: /local/disk1/pmandl/01_Software/FLOWExaFMM/src/fmm.so: undefined symbol: _ZN5jlcxx6Module12get_constantERKSs ERROR: LoadError: Failed to precompile FLOWExaFMM [a07d1f4e-0e34-4d8b-bfef-e5b961477d34] to /home/pmandl/.julia/compiled/v1.6/FLOWExaFMM/jl_T5D5wX. Stacktrace: [1] error(s::String) @ Base ./error.jl:33 [2] compilecache(pkg::Base.PkgId, path::String, internal_stderr::Base.TTY, internal_stdout::Base.TTY, ignore_loaded_modules::Bool) @ Base ./loading.jl:1385 [3] compilecache(pkg::Base.PkgId, path::String) @ Base ./loading.jl:1329 [4] _require(pkg::Base.PkgId) @ Base ./loading.jl:1043 [5] require(uuidkey::Base.PkgId) @ Base ./loading.jl:936 [6] require(into::Module, mod::Symbol) @ Base ./loading.jl:923 [7] include(fname::String) @ Base.MainInclude ./client.jl:444 [8] top-level scope @ none:6 in expression starting at /local/disk1/pmandl/01_Software/FLOWExaFMM/test/runtests.jl:2 ERROR: Package FLOWExaFMM errored during testing
> 
> _do you have any ideas which might help? Thank you very much for your support!_

**Comment by EdoAlvarezR** [comment]
Are you following the instructions in the docs? [https://flow.byu.edu/FLOWUnsteady/installation/general/#FLOWVPM](https://flow.byu.edu/FLOWUnsteady/installation/general/#FLOWVPM)

As indicated in the docs:
> Test that you can correctly compile C++ code wrapped for Julia following these instructions: [LINK](https://nbviewer.org/github/byuflowlab/FLOWVPM.jl/blob/master/docs/installation-linux.ipynb)

**Comment by philippmandl** [comment]
yes, I followed both instructions and the C++ wrapper works correctly - all the tests are passed. I have tried it with Juli 1.6.7, 1.8.5 and 1.9.1 - but always the same problem at the scientific linux system - do you have any ideas?

**Comment by EdoAlvarezR** [comment]
Can you please confirm that the file `/local/disk1/pmandl/01_Software/FLOWExaFMM/src/fmm.so` exists? If not, the compilation didn't work.

**Comment by philippmandl** [comment]
Yes, this file does exist - that's why I am wondering, why the Test fails.

**Comment by philippmandl** [comment]
when running the precompiler, following error is still present 

julia> import Pkg; Pkg.precompile()
Precompiling project...
  ✗ FLOWExaFMM
  ✗ FLOWVPM
  66 dependencies successfully precompiled in 29 seconds

ERROR: The following 2 direct dependencies failed to precompile:

FLOWExaFMM [a07d1f4e-0e34-4d8b-bfef-e5b961477d34]

Failed to precompile FLOWExaFMM [a07d1f4e-0e34-4d8b-bfef-e5b961477d34] to /home/pmandl/.julia/compiled/v1.6/FLOWExaFMM/jl_Nn0DUu.
/local/disk1/pmandl/01_Software/julia-1.6.7/bin/julia: symbol lookup error: /local/disk1/pmandl/01_Software/FLOWExaFMM/src/fmm.so: undefined symbol: _ZN5jlcxx6Module12get_constantERKSs

FLOWVPM [6e19019d-7c31-4940-9d16-c3f15dfe6020]

Failed to precompile FLOWVPM [6e19019d-7c31-4940-9d16-c3f15dfe6020] to /home/pmandl/.julia/compiled/v1.6/FLOWVPM/jl_EC2Pyj.
/local/disk1/pmandl/01_Software/julia-1.6.7/bin/julia: symbol lookup error: /local/disk1/pmandl/01_Software/FLOWExaFMM/src/fmm.so: undefined symbol: _ZN5jlcxx6Module12get_constantERKSs
ERROR: LoadError: Failed to precompile FLOWExaFMM [a07d1f4e-0e34-4d8b-bfef-e5b961477d34] to /home/pmandl/.julia/compiled/v1.6/FLOWExaFMM/jl_8ibDQS.
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:33
  [2] compilecache(pkg::Base.PkgId, path::String, internal_stderr::IOContext{Base.PipeEndpoint}, internal_stdout::IOContext{IOStream}, ignore_loaded_modules::Bool)
    @ Base ./loading.jl:1385
  [3] compilecache(pkg::Base.PkgId, path::String)
    @ Base ./loading.jl:1329
  [4] _require(pkg::Base.PkgId)
    @ Base ./loading.jl:1043
  [5] require(uuidkey::Base.PkgId)
    @ Base ./loading.jl:936
  [6] require(into::Module, mod::Symbol)
    @ Base ./loading.jl:923
  [7] include
    @ ./Base.jl:384 [inlined]
  [8] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::Nothing)
    @ Base ./loading.jl:1235
  [9] top-level scope
    @ none:1
 [10] eval
    @ ./boot.jl:360 [inlined]
 [11] eval(x::Expr)
    @ Base.MainInclude ./client.jl:446
 [12] top-level scope
    @ none:1
in expression starting at /home/pmandl/.julia/packages/FLOWVPM/WA9ug/src/FLOWVPM.jl:1

Stacktrace:
 [1] pkgerror(msg::String)
   @ Pkg.Types /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/Types.jl:55
 [2] precompile(ctx::Pkg.Types.Context; internal_call::Bool, strict::Bool, warn_loaded::Bool, kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ Pkg.API /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/API.jl:1265
 [3] precompile
   @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/API.jl:921 [inlined]
 [4] #precompile#196
   @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/API.jl:919 [inlined]
 [5] precompile()
   @ Pkg.API /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/API.jl:919
 [6] top-level scope
   @ REPL[7]:1


do you have any ideas regarding this problem? I am running out of ideas currently

**Comment by philippmandl** [comment]
running test FLOWExaFMM leads to the error

Testing FLOWExaFMM
      Status `/tmp/jl_MB0FLT/Project.toml`
  [1f15a43c] CxxWrap v0.11.2
  [a07d1f4e] FLOWExaFMM v2.1.0 `/local/disk1/pmandl/01_Software/FLOWExaFMM`
  [8dfed614] Test `@stdlib/Test`
      Status `/tmp/jl_MB0FLT/Manifest.toml`
  [1f15a43c] CxxWrap v0.11.2
  [a07d1f4e] FLOWExaFMM v2.1.0 `/local/disk1/pmandl/01_Software/FLOWExaFMM`
  [692b3bcd] JLLWrappers v1.4.1
  [1914dd2f] MacroTools v0.5.10
  [21216c6a] Preferences v1.4.0
  [3eaa8342] libcxxwrap_julia_jll v0.8.8+1
  [0dad84c5] ArgTools `@stdlib/ArgTools`
  [56f22d72] Artifacts `@stdlib/Artifacts`
  [2a0f44e3] Base64 `@stdlib/Base64`
  [ade2ca70] Dates `@stdlib/Dates`
  [f43a241f] Downloads `@stdlib/Downloads`
  [b77e0a4c] InteractiveUtils `@stdlib/InteractiveUtils`
  [b27032c2] LibCURL `@stdlib/LibCURL`
  [76f85450] LibGit2 `@stdlib/LibGit2`
  [8f399da3] Libdl `@stdlib/Libdl`
  [56ddb016] Logging `@stdlib/Logging`
  [d6f4376e] Markdown `@stdlib/Markdown`
  [ca575930] NetworkOptions `@stdlib/NetworkOptions`
  [44cfe95a] Pkg `@stdlib/Pkg`
  [de0858da] Printf `@stdlib/Printf`
  [3fa0cd96] REPL `@stdlib/REPL`
  [9a3f8284] Random `@stdlib/Random`
  [ea8e919c] SHA `@stdlib/SHA`
  [9e88b42a] Serialization `@stdlib/Serialization`
  [6462fe0b] Sockets `@stdlib/Sockets`
  [fa267f1f] TOML `@stdlib/TOML`
  [a4e569a6] Tar `@stdlib/Tar`
  [8dfed614] Test `@stdlib/Test`
  [cf7118a7] UUIDs `@stdlib/UUIDs`
  [4ec0a83e] Unicode `@stdlib/Unicode`
  [deac9b47] LibCURL_jll `@stdlib/LibCURL_jll`
  [29816b5a] LibSSH2_jll `@stdlib/LibSSH2_jll`
  [c8ffd9c3] MbedTLS_jll `@stdlib/MbedTLS_jll`
  [14a3606d] MozillaCACerts_jll `@stdlib/MozillaCACerts_jll`
  [83775a58] Zlib_jll `@stdlib/Zlib_jll`
  [8e850ede] nghttp2_jll `@stdlib/nghttp2_jll`
  [3f19e933] p7zip_jll `@stdlib/p7zip_jll`
Precompiling project...
  ✗ FLOWExaFMM
  0 dependencies successfully precompiled in 1 seconds (8 already precompiled)
  1 dependency errored. To see a full report either run `import Pkg; Pkg.precompile()` or load the package
     Testing Running tests...
/local/disk1/pmandl/01_Software/julia-1.6.7/bin/julia: symbol lookup error: /local/disk1/pmandl/01_Software/FLOWExaFMM/src/fmm.so: undefined symbol: _ZN5jlcxx6Module12get_constantERKSs
ERROR: LoadError: Failed to precompile FLOWExaFMM [a07d1f4e-0e34-4d8b-bfef-e5b961477d34] to /home/pmandl/.julia/compiled/v1.6/FLOWExaFMM/jl_xaHY8X.
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:33
 [2] compilecache(pkg::Base.PkgId, path::String, internal_stderr::Base.TTY, internal_stdout::Base.TTY, ignore_loaded_modules::Bool)
   @ Base ./loading.jl:1385
 [3] compilecache(pkg::Base.PkgId, path::String)
   @ Base ./loading.jl:1329
 [4] _require(pkg::Base.PkgId)
   @ Base ./loading.jl:1043
 [5] require(uuidkey::Base.PkgId)
   @ Base ./loading.jl:936
 [6] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:923
 [7] include(fname::String)
   @ Base.MainInclude ./client.jl:444
 [8] top-level scope
   @ none:6
in expression starting at /local/disk1/pmandl/01_Software/FLOWExaFMM/test/runtests.jl:2
ERROR: Package FLOWExaFMM errored during testing

**Comment by EdoAlvarezR** [comment]
Is the Hello World test mentioned here in the manual compilation passing? https://nbviewer.org/github/byuflowlab/FLOWVPM.jl/blob/master/docs/installation-linux.ipynb
